scene/gui/rich_text_effect.cpp

```diff
@@ -0,0 +1,90 @@
+#include "rich_text_effect.h"
+
+#include "core/script_language.h"
+
+void RichTextEffect::_bind_methods() {
+	BIND_VMETHOD(MethodInfo(Variant::INT, "_process_custom_fx", PropertyInfo(Variant::OBJECT, "char_fx", PROPERTY_HINT_RESOURCE_TYPE, "CustomFXChar")));
+}
+
+Variant RichTextEffect::get_bbcode() const {
+	Variant r;
+	if (get_script_instance()) {
+		if (!get_script_instance()->get("bbcode", r)) {
+			String path = get_script_instance()->get_script()->get_path();
+			r = path.get_file().get_basename();
+		}
+	}
+	return r;
+}
+
+bool RichTextEffect::_process_effect_impl(Ref<CharFXTransform> cfx) {
+	Variant v = false;
+	if (get_script_instance()) {
+		v = get_script_instance()->call("_process_custom_fx", cfx);
+		if (v.get_type() != Variant::BOOL) {
+			v = false;
+		}
+	}
+	return v;
+}
+
+RichTextEffect::RichTextEffect() {
+}
+
+void CharFXTransform::_bind_methods() {
+
+	ClassDB::bind_method(D_METHOD("get_relative_index"), &CharFXTransform::get_relative_index);
+	ClassDB::bind_method(D_METHOD("set_relative_index", "i"), &CharFXTransform::set_relative_index);
+
+	ClassDB::bind_method(D_METHOD("get_absolute_index"), &CharFXTransform::get_absolute_index);
+	ClassDB::bind_method(D_METHOD("set_absolute_index", "i"), &CharFXTransform::set_absolute_index);
+
+	ClassDB::bind_method(D_METHOD("get_elapsed_time"), &CharFXTransform::get_elapsed_time);
+	ClassDB::bind_method(D_METHOD("set_elapsed_time", "t"), &CharFXTransform::set_elapsed_time);
+
+	ClassDB::bind_method(D_METHOD("is_visible"), &CharFXTransform::is_visible);
+	ClassDB::bind_method(D_METHOD("set_visibility", "vis"), &CharFXTransform::set_visibility);
```

Comment:
Please use fully qualified names for arguments, not abbreviations. This should be `visible`.